ramp/post_process/post_process.py

```diff
@@ -60,17 +59,618 @@ def Profile_series_plot(stoch_profiles_series):
+        """initializing a Plot object from a file results
+
+        Parameters
+        ----------
+        file : str
+            path to the file of the results
+        sheet_name : int,str, optional
+            sheet_name of the result in case an excel file is passed, by default 0
+        sep : str, optional
+            separator in case a csv file is passed, by default ","
+        index: pd.DatetimeIndex, optional
+            if df index is not pd.DatetimeIndex, an DatetimeIndex object can be passed to change the index
+
+        Returns
+        -------
+        Plot
+            A Plot object
+        """
+
+        if file.split(".")[-1] == "csv":
+            df = pd.read_excel(file,sheet_name=sheet_name,index_col=0,header=0,sep=sep)
+
+        elif file.split(".")[-1] == "xlsx":
+
+            df = pd.read_excel(file,sheet_name=sheet_name,index_col=0,header=0)
+
+        return Plot(df,index)
+
+
+
+    def __init__(self,df,index=None):
+        """initializes a Plot object using a pd.DataFrame
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            a pd.DataFrame with pd.DatetimeIndex
+
+        index: pd.DatetimeIndex, optional
+            if df index is not pd.DatetimeIndex, an DatetimeIndex object can be passed to change the index
+        """
+        if index is not None:
+            df = df.copy()
+            df.index = index
+
+        self.DataFrame = df
+
+
+    @property
+    def freq(self):
+        """return the frequency of the pd.DatetimeIndex
+        """
+        return self.df.index.freq
+
+    @property
+    def columns(self):
+        """returns a list of columns (cases) of the Plot dataframe
+
+        Returns
+        -------
+        list
+            list of cases
+        """
+        return self.df.columns.tolist()
+
+    @columns.setter
+    def columns(self,var):
+        self.df.columns = var
+
+    @property
+    def index(self):
+        return self.df.index
+
+    def resample(self,freq,rule,conversion="*1",inplace=False):
+        """returns a resampled version of the data
+
+        Parameters
+        ----------
+        freq : str
+            pd.DataFrame.resample frequency, for example: "1h" for hourly resampling, "1w" for weekly data. Refer to https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.resample.html
+        rule : str
+            resampling rule. acceptable values are 'sum', 'mean', 'nearest', ...
+        conversion : str, optional
+            if resampling needs a conversion, can be done through this argument, by default "*1"
+        inplace : bool, optional
+            if True, implements the changes inplace otherwise, returns the resampled data as a new object, by default False
+
+        Returns
+        -------
+        Plot, None
+            Plot object if inplace=False otherwise returns None
+        """
+        df = self.df.copy()
+
+        df = eval(f"df{conversion}")
```

Comment:
```suggestion
        df = df * conversion
```
Using `eval` can lead to safety issues, it is considered good practice not to use it in code that you publish. As you only use it once, there is certainly a way around using eval we can find :)

Reply:
Sure!